### DISCUSSION TITLE
Picking cells with offscreen rendering

### DISCUSSION THREAD

**Discussion by ghost** [pyvista/pyvista#4352] [Q&A] [answered]
Hello everyone !

I have been playing a lot with the cell picking option offered by the plotter, and with it's underlying OpenGL selector, it works really well and fast when a GPU is present.

Now, basically, what I need to do in my current project is something akin to occlusion culling, where I need to tag only cells of a polydata / unstructured grid that are visible from a given camera position, angle, etc... 
I realized that the implementation in `plotting/picking.py` could really help me with this but since I do not need all the actual displaying on screen, I would love to make it work with "offscreen" rendering and a picker that reads in the buffer rather than the screen pixels ...

@banesullivan I could read a lot of issues and PRs where you handled this kind of work, would you by any chance have any insights in how I could achieve this ? If necessary, I don't mind doing C++ either ...

Thank you all for your help !
Cheers,

Thibault

**Comment by banesullivan** [comment, marked as answer]
If yor goal is offscreen picking, maybe take a look at our testing of the picking where we do this: https://github.com/pyvista/pyvista/blob/main/tests/test_picking.py

Hope this helps!

**Reply by ghost** [reply]
Ah thank you @banesullivan, this indeed does the trick for now ! :)
